@here/olp-sdk-dataservice-read/lib/DataStoreRequestBuilder.ts

```diff
@@ -32,10 +32,7 @@ async function addBearerToken(
      * @todo fix datastore-api template generator.
      * It sends empty object for headers, but should be Headers object
      */
-    if (
-        init.headers === undefined ||
-        Object.entries(init.headers).length === 0
-    ) {
+    if (init.headers === undefined || !(init.headers instanceof Headers)) {
```

Comment:
what about if init.headers will be an empty object?

Reply:
Than we'll replace it with Headers instance. Nothing will be lose